### PR TITLE
Fix missing wfs_config and use .tif extension for rasters

### PR DESCRIPTION
**Pull request by ericboucher**
Better error handling around geotiffs + fix a small issue with missing wcs_config arg in the demo endpoint

### CODE REVIEW

File: api-flask/app/caching.py [hunk 1]
```diff
@@ -24,16 +28,33 @@ def cache_file(url, prefix, extension='cache'):
...
     if os.path.isfile(cache_filepath):
         logger.info('Returning cached file for {}.'.format(url))
-        return cache_filepath
+
+        # if the file is a geotiff, confirm that we can open it.
+        if extension == 'tif':
+            try:
+                rasterio.open(cache_filepath)
+                return cache_filepath
+            except Exception:
```
Comment (JorgeMartinezG) [inline api-flask/app/caching.py:37]: We would need to specify the type of exception we are silencing. For this case, we could wrap all possible errors from rasterio with **rasterio.errors.RasterioError** or *rasterio.errors.RasterioIOError*

File: api-flask/app/caching.py [hunk 2]
```diff
@@ -24,16 +28,33 @@ def cache_file(url, prefix, extension='cache'):
...
+                pass
+        else:
+            return cache_filepath
+
     # If the file does not exist, download and return path.
-    else:
-        r = requests.get(url, verify=False)
+    response = requests.get(url, verify=False)
+
+    try:
```
Comment (JorgeMartinezG) [inline api-flask/app/caching.py:45]: For this case, we could validate the status code we are receiving and then specify the status code we are getting from the remote server. Something like

```
if response.status_code != 200:
  raise InternalServer.. 
```

File: api-flask/app/zonal_stats.py [hunk 3]
```diff
@@ -161,13 +163,17 @@ def calculate_stats(
...
-    )
+    try:
+        stats = zonal_stats(
+            stats_input,
+            geotiff,
+            stats=stats,
+            prefix=prefix,
+            geojson_out=geojson_out
+        )
+    except Exception as e:
```
Comment (JorgeMartinezG) [inline api-flask/app/zonal_stats.py:174]: Similar to the previous comment. We could wrap this exception into a RasterioError one.

**Review by JorgeMartinezG — CHANGES_REQUESTED** [review]
Added a few comments related on how the errors are handled in this PR.

File: api-flask/app/caching.py (hunk 2 above)
Comment (ericboucher) [reply api-flask/app/caching.py:45]: The error status will be specified in the error already. I switched to using the HTTPError exception

**Review by JorgeMartinezG — APPROVED** [review]

**Pull request merged**